src/WalletDerive.php

```diff
@@ -194,6 +194,53 @@ function serializePrivKey($symbol, $network, $key) {
         return $hex ? '0x' . $key->getHex() : $key->toWif($network);
     }
 
+    /**
+     * Get extended public keys from a given key in all available formats
+     */
+    function getExtendedPublicKeys($key) {
+        $params = $this->get_params();
+        $coin = $params['coin'];
+        list($symbol) = explode('-', $coin);
+
+        $networkCoinFactory = new NetworkCoinFactory();
+        $network = $networkCoinFactory->getNetworkCoinInstance($coin);
+        Bitcoin::setNetwork($network);
+
+        // get initial key type for the coin
+        $initial_key_type = $this->getKeyTypeFromCoinAndKey($coin, $key);
+
+        // store results here
+        $extkeys = array();
+
```

Comment:
May I suggest you arrange these pieces of code as a function e.g. generate_xyzpub somehow like this:
```$extkeys = generate_xyzpub()```


```
function generate_xyzpub()
{
    $addrTypes = [
        'x' => 'legacy',
        'y' => 'p2sh-segwit',
        'z' => 'bech32'
    ]
    $types = ['x', 'y', 'z'];
    foreach ($types as $type) {
        $this->params['addr-type'] = $addrTypes[$type];
        $key_type = $type;
        if ( $this->networkSupportsKeyType(.....
       .........
    }
    return $extkeys;
}

```

I would also suggest you [squash](https://github.com/wprig/wprig/wiki/How-to-squash-commits) all the commits into a single one to keep nice clean history

Reply:
yes I concur with @melaxon's suggestions.